packages/Makefile

```diff
@@ -0,0 +1,233 @@
+.DEFAULT_GOAL := all
+SHELL ?= /bin/bash
+
+# Get the parent directory of this Makefile
+# WARNING: `lastword` will not work if there are spaces in this Makefile path
+BASE_DIR := $(abspath $(dir $(abspath $(lastword $(MAKEFILE_LIST)))))
+
+# Set the build and ISO root if this Makefile was invoked directly
+BUILD_ROOT ?= $(abspath $(BASE_DIR)/../_build)
+ISO_ROOT ?= $(BUILD_ROOT)/root
+
+#
+# Docker builder
+#
+
+BUILDER_IMAGE ?= metalk8s-build:latest
+BUILDER_CONTAINER := $(BUILD_ROOT)/package-build-container
+
+# This target is a simple empty file used to track whether the docker image is
+# up-to-date, and can be used as an order-only prerequisite
+$(BUILDER_CONTAINER): Dockerfile entrypoint.sh
+	mkdir -p $(@D)
+	rm -f $@
+	docker build -t $(BUILDER_IMAGE) -f $< $(dir $<)
+	touch $@
+
+DOCKER ?= docker
+
+TARGET_UID := $(shell id -u)
+TARGET_GID := $(shell id -g)
+
+#
+# Packages
+#
+
+PACKAGES_ROOT := $(BUILD_ROOT)/packages
+$(PACKAGES_ROOT):
+	mkdir -p $@
+
+# Sub-directories to create for each package
+PACKAGE_SUBDIRS := SOURCES SRPMS RPMS SPECS
+
+# Templated rules for building a RPM from sources (see packages/packages.mk for
+# an example usage)
+define create-pkg-from-source =
```

Comment:
My .02: once starting to use template rules and functions and macros and whatnot in `Makefile`s, we should stop using `make`.

Reply:
Agreed, which is why I left the PR in draft state. The main goal is to show how bad it can become, in comparison to more powerful solutions (e.g. `doit`, incoming PR from @slaperche-scality)